libraries/nn_idb/src/IDBReader.cpp

```diff
@@ -83,7 +83,7 @@ namespace nn::idb {
 
     bool IDBReader::Initialize() {
         uint64_t menuTid = _SYSGetSystemApplicationTitleId(SYSTEM_APP_ID_WII_U_MENU);
-        auto *menuTidPtr = (uint32_t *) menuTid;
+        auto *menuTidPtr = (uint32_t *) &menuTid;
         char path[90];
         snprintf(path, sizeof(path), "fs:/vol/storage_mlc01/usr/save/%08X/%08X/user/common/BaristaIconDataBase.dat", menuTidPtr[0], menuTidPtr[1]);
```

Comment:
I'd probably just do something like this instead of casting the TID to a 32-bit pointer.
```suggestion
        char path[90];
        snprintf(path, sizeof(path), "fs:/vol/storage_mlc01/usr/save/%08X/%08X/user/common/BaristaIconDataBase.dat", (uint32_t) (menuTid >> 32), (uint32_t) (menuTid & 0xffffffff));
```